docs/docs/gatsby-image-api.md

```diff
@@ -0,0 +1,344 @@
+---
+title: Gatsby Image API
+---
+
+Part of what makes Gatsby sites so fast is its recommended approach to handling images. `gatsby-image` is a React component designed to work seamlessly with Gatsby’s [native image processing](https://image-processing.gatsbyjs.org/) capabilities powered by GraphQL and [gatsby-plugin-sharp](/packages/gatsby-plugin-sharp/) to easily and completely optimize image loading for your sites.
+
+> _Warning: gatsby-image is **not** a drop-in replacement for `<img />`. It’s optimized for fixed width/height images and images that stretch the full-width of a container._
+
+Demo: [https://using-gatsby-image.gatsbyjs.org/](https://using-gatsby-image.gatsbyjs.org/)
+
+## In this doc
+
+- [Setting up Gatsby Image](#setting-up-gatsby-image)
+- [Types of gatsby-images](#types-of-gatsby-images)
+  - [Fixed images and parameters](#images-with-a-fixed-width-and-height)
+  - [Fluid images and parameters](#images-that-stretch-across-a-fluid-container)
+  - [Resized images](#resized-images)
+  - [Shared query parameters](#shared-query-parameters)
+- [Image query fragments](#image-query-fragments)
+- [Gatsby Image props](#gatsby-image-props)
+
+## Setting up Gatsby Image
+
+To install Gatsby Image, install the `gatsby-image` package along with necessary plugins `gatsby-transformer-sharp` and `gatsby-plugin-sharp`. Reference the packages in your Gatsby config file. You can also provide additional options to [`gatsby-plugin-sharp`](/packages/gatsby-plugin-sharp/) in your config file.
```

Comment:
Should we include a copy-pasteable block here for convenience?

Reply:
I debated on that. It's included so many places, I linked to the docs instead. But we could add it if necessary

Reply:
I ended up adding it in! Thanks for the nudge.